modules/hyprpaper/hm.nix

```
@@ -1,10 +1,17 @@
 { config, lib, ... }:
 {
-  options.stylix.targets.hyprpaper.enable =
-    config.lib.stylix.mkEnableTarget "Hyprpaper" true;
+  options.stylix.targets.hyprpaper = {
+    enable = config.lib.stylix.mkEnableTarget "Hyprpaper" true;
+    wallpaper = config.lib.stylix.mkEnableWallpaper "Hyprpaper" true;
+  };
 
   config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.hyprpaper.enable)
+    lib.mkIf
+      (
+        config.stylix.enable
+        && config.stylix.targets.hyprpaper.enable
+        && config.stylix.targets.hyprpaper.wallpaper
+      )
```

Although the [`lib.mkIf (config.stylix.enable && config.stylix.targets.<TARGET>.enable)` pattern is currently not consistent in the code base](https://github.com/danth/stylix/issues/400), it might be better to:

```suggestion
    lib.mkIf (config.stylix.enable && config.stylix.targets.hyprpaper.enable)
```

and add the guard inside instead:

```nix
{
  services.hyprpaper.settings =
    lib.mkIf config.stylix.targets.hyprpaper.wallpaper {
      preload = [ "${config.stylix.image}" ];
      wallpaper = [ ",${config.stylix.image}" ];
    };
};
```